src/framework/components/model/component.js

```diff
@@ -24,65 +26,121 @@ import { Component } from '../component.js';
-        super(system, entity);
+    _type = 'asset';
 
-        this._type = 'asset';
-
-        /**
-         * @type {number}
-         * @private
-         */
-        this._asset = null;
-        /**
-         * @type {Model}
-         * @private
-         */
-        this._model = null;
-
-        this._mapping = {};
-
-        this._castShadows = true;
-        this._receiveShadows = true;
-
-        /**
-         * @type {Asset}
-         * @private
-         */
-        this._materialAsset = null;
-        this._material = system.defaultMaterial;
+    /**
+     * @type {Asset|number|null}
+     * @private
+     */
+    _asset = null;
 
-        this._castShadowsLightmap = true;
-        this._lightmapped = false;
-        this._lightmapSizeMultiplier = 1;
-        this._isStatic = false;
+    /**
+     * @type {Model|null}
+     * @private
+     */
+    _model = null;
 
-        this._layers = [LAYERID_WORLD]; // assign to the default world layer
-        this._batchGroupId = -1;
+    /* eslint-disable jsdoc/check-types */
+    /**
+     * @type {Object.<string, number>}
+     * @private
+     */
+    _mapping = {};
+    /* eslint-enable jsdoc/check-types */
 
-        /**
-         * @type {BoundingBox}
-         * @private
-         */
-        this._customAabb = null;
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _castShadows = true;
 
-        this._area = null;
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _receiveShadows = true;
 
-        this._assetOld = 0;
-        this._materialEvents = null;
-        this._dirtyModelAsset = false;
-        this._dirtyMaterialAsset = false;
+    /**
+     * @type {Asset|number|null}
+     * @private
+     */
+    _materialAsset = null;
+
+    /**
+     * @type {Material}
+     * @private
+     */
+    _material;
+
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _castShadowsLightmap = true;
 
-        this._clonedModel = false;
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _lightmapped = false;
 
-        // #if _DEBUG
-        this._batchGroup = null;
-        // #endif
+    /**
+     * @type {number}
+     * @private
+     */
+    _lightmapSizeMultiplier = 1;
+
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _isStatic = false;
+
+    /**
+     * @type {number[]}
+     * @private
+     */
+    _layers = [LAYERID_WORLD]; // assign to the default world layer
+
+    /**
+     * @type {number}
+     * @private
+     */
+    _batchGroupId = -1;
+
+    /**
+     * @type {BoundingBox|null}
+     * @private
+     */
+    _customAabb = null;
+
+    _area = null;
+
+    _materialEvents = null;
+
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _clonedModel = false;
+
+    // #if _DEBUG
+    _batchGroup = null;
+    // #endif
+
+    /**
+     * Create a new ModelComponent instance.
+     *
+     * @param {ModelComponentSystem} system - The ComponentSystem that created this Component.
```

Comment:
should constructor in these be public?

Reply:
I was thinking about this already. We should add `@hideconstructor` for all components. But I think it's best to do that in a dedicated follow up PR, since this one is all about the `ModelComponent`.